src/main/java/ca/wolfram/beta/symath/MathUtils.java

```diff
@@ -19,4 +24,31 @@ public static void print(String s, Object... o) {
         System.out.println(String.format(Locale.CANADA, s, o));
     }
 
+    public static Set<MathNode> getVariableTerms(MathNode node) {
+        Set<MathNode> set;
+        switch (node.getType()) {
+        case CONSTANT:
+        case MATH_CONSTANT:
+            set = new HashSet<MathNode>();
+            break;
+        case VARIABLE:
+            set = new HashSet<MathNode>();
+            set.add(node);
+            break;
+        default:
+            set = node.getChildren().stream().filter(children -> !children.isConstant()).collect(Collectors.toSet());
```

Comment:
(x + 1)^2. I'm not sure how far we'll be going to simplify the nodes, but that could result in some constants being carried over.

Also, 2x + 3y is simplified, but without recursion, you'll just see the ADD type and add all those values into your set because both multiply nodes aren't constants